iceoryx_posh/test/moduletests/test_popo_event_variable.cpp

```diff
@@ -0,0 +1,247 @@
+}
+
+TEST_F(EventVariable_test, WaitAndNotifyResultsInCorrectNotificationVector)
+{
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 5U;
+    EventNotifier notifier(m_eventVarData, index);
+    EventListener listener(m_eventVarData);
+    notificationVector activeNotifications;
+
+    std::thread waiter([&] {
+        activeNotifications = listener.wait();
+        ASSERT_THAT(activeNotifications.size(), Eq(1U));
+        EXPECT_THAT(activeNotifications[0], Eq(index));
+    });
+
+    notifier.notify();
+    waiter.join();
+}
+
+TIMING_TEST_F(EventVariable_test, WaitBlocks, Repeat(5), [&] {
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 5U;
+    EventNotifier notifier(m_eventVarData, index);
+    EventListener listener(m_eventVarData);
+    notificationVector activeNotifications;
+    iox::posix::Semaphore semaphore =
+        iox::posix::Semaphore::create(iox::posix::CreateUnnamedSingleProcessSemaphore, 0U).value();
+    std::atomic_bool hasWaited{false};
+
+    std::thread waiter([&] {
+        semaphore.post();
+        activeNotifications = listener.wait();
+        hasWaited.store(true, std::memory_order_relaxed);
+        ASSERT_THAT(activeNotifications.size(), Eq(1U));
+        EXPECT_THAT(activeNotifications[0], Eq(index));
+    });
+
+    semaphore.wait();
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    EXPECT_THAT(hasWaited, Eq(false));
+    notifier.notify();
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    EXPECT_THAT(hasWaited, Eq(true));
+    waiter.join();
+})
+
+TIMING_TEST_F(EventVariable_test, SecondWaitBlocksUntilNewNotification, Repeat(5), [&] {
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 2U;
+    EventNotifier notifier1(m_eventVarData, index);
+    EventNotifier notifier2(m_eventVarData, 0U);
+    EventListener listener(m_eventVarData);
+    iox::posix::Semaphore semaphore =
+        iox::posix::Semaphore::create(iox::posix::CreateUnnamedSingleProcessSemaphore, 0U).value();
+    std::atomic_bool hasWaited{false};
+
+    notifier1.notify();
+    notifier2.notify();
+    notificationVector activeNotifications = listener.wait();
+    ASSERT_THAT(activeNotifications.size(), Eq(2U));
+    EXPECT_THAT(activeNotifications[0], Eq(0U));
+    EXPECT_THAT(activeNotifications[1], Eq(index));
+
+    std::thread waiter([&] {
+        semaphore.post();
+        activeNotifications = listener.wait();
+        hasWaited.store(true, std::memory_order_relaxed);
+        ASSERT_THAT(activeNotifications.size(), Eq(1U));
+        EXPECT_THAT(activeNotifications[0], Eq(index));
+    });
+
+    semaphore.wait();
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    EXPECT_THAT(hasWaited, Eq(false));
+    notifier1.notify();
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    EXPECT_THAT(hasWaited, Eq(true));
+    waiter.join();
+})
+
+TEST_F(EventVariable_test, AllEntriesAreResetToFalseInsideWait)
```

Comment:
isn't this essentially already checked by `SecondWaitBlocksUntilNewNotification`